src/user/user.service.spec.ts

```diff
@@ -1,36 +1,78 @@
-import { UserModule } from "./user.module";
+
 import { Test, TestingModule } from "@nestjs/testing";
-import { UserService } from "./user.service";
-import { expect } from "chai";
-import { DatabaseModule } from "../database/database.module";
-import { userProviders } from "./user.providers";
 import { ConfigModule } from "@nestjs/config";
 import { IUserService } from "./user.types";
+import { ProviderTokens } from "../tokens";
+import { Repository } from "typeorm";
+import { User } from "../database/entities/user.entity";
+import { UserService } from "./user.service";
+
+const userRepositoryMock = {
+    findOne: (): Promise<User> => {
+        return new Promise((resolve, reject) => {
+            const result: User = {
+                id: "1",
+                email: "test@test.com",
+                createdAt: new Date(),
+                updatedAt: new Date(),
+            }
+            resolve(result);   
+        }); 
+    },
+    find: (): Promise<User[]> => {
+        return new Promise((resolve, reject) => {
+            const result: User[] =[
+                {
+                    id: "1",
+                    email: "test@test.com",
+                    createdAt: new Date(),
+                    updatedAt: new Date(),
+                },
+                {
+                    id: "2",
+                    email: "test2@test.com",
+                    createdAt: new Date(),
+                    updatedAt: new Date(),
+                }
+            ]
+            resolve(result);
+        });
+    }
+};
 
 describe("UserService", () => {
     let service: IUserService;
 
     beforeEach(async () => {
         const module: TestingModule = await Test.createTestingModule({
             imports: [
-                DatabaseModule,
-                UserModule,
                 ConfigModule.forRoot({
                     isGlobal: true
-                })
-            ],
-            providers: [...userProviders, UserService]
+                }),
+            ],          
+            providers: [
+                {
+                    provide: ProviderTokens.UserService,
+                    useFactory: () => {
+                        return new UserService(userRepositoryMock as unknown as Repository<User>);
```

Comment:
I think you want to `useValue` here so you can mock test-specific values.

Alternatively, and this might be better, actually, is to use the `module.get<UserService, jest.Mocked<UserService>>( ProviderTokens.UserService );` to populate a ``let mockedUserService: jest.Mocked<UserService>;` - you'd get a fresh instance for each test that way and it would be a lot harder to cross-polute tests.